### DISCUSSION TITLE
大佬不考虑做个离线版吗？

### DISCUSSION THREAD

**Discussion by yanhuacuo** [pdog18/Soak#3] [General]
解析 yaml 目前最优解可能就是使用 https://github.com/jbeder/yaml-cpp 这个项目，大佬可以考虑做个通用版。

目前，我们的实现是基于文本模式的读改，只能局限于特定的方案：

https://wubi98.gitee.io/2023/02/21/2023-02-21-01.rimetool/

WIKI：https://gitee.com/wubi98/shell/wikis/RIME%20%E7%9B%AE%E5%BD%95%E7%9B%B4%E8%BE%BE

如果考虑做基于 C 或 C++ 的离线版，我可以提供一些帮助。

**Comment by pdog18** [comment]
> 不考虑做个离线版吗？

离线版将使用 Electron App 来制作。

> 解析 yaml 目前最优解可能就是使用 https://github.com/jbeder/yaml-cpp 这个项目

感谢您的推荐，请问一下 yaml-cpp 相对于别的 yaml 的优势是什么呀？因为我正在使用 TypeScript 来开发，所以使用的是 [yaml](https://github.com/eemeli/yaml) 这个库。

> 可以考虑做个通用版。

做通用版的意思是什么？对于 Rime 在不同平台的通用版还是说对于本项目在不同平台的通用版？



> 目前，我们的实现是基于文本模式的读改，只能局限于特定的方案：
> https://wubi98.gitee.io/2023/02/21/2023-02-21-01.rimetool/

惭愧惭愧，我在开发本项目之前，居然都不知道有这样一个好的项目，我学习一下！

> 如果考虑做基于 C 或 C++ 的离线版，我可以提供一些帮助。

太感谢了！绝无冒犯之意，但是我不太明白需要做成 C / C++ 的离线版相对于 TypeScript(web-app/electron-app) 的优势是在哪里？ 

希望以后可以与您多多交流！感谢！

**Comment by pdog18** [comment]
> 解析 yaml 目前最优解可能就是使用 https://github.com/jbeder/yaml-cpp 这个项目，大佬可以考虑做个通用版。
> 
> 目前，我们的实现是基于文本模式的读改，只能局限于特定的方案：
> 
> https://wubi98.gitee.io/2023/02/21/2023-02-21-01.rimetool/
> 
> WIKI：https://gitee.com/wubi98/shell/wikis/RIME%20%E7%9B%AE%E5%BD%95%E7%9B%B4%E8%BE%BE
> 
> 如果考虑做基于 C 或 C++ 的离线版，我可以提供一些帮助。

我浏览了您的 repo，您说的通用版指的是不仅「Rime」也包括其他一些输入法也能利用的「通用」吗？

**Comment by yanhuacuo** [comment]
> > 解析 yaml 目前最优解可能就是使用 https://github.com/jbeder/yaml-cpp 这个项目，大佬可以考虑做个通用版。
> > 目前，我们的实现是基于文本模式的读改，只能局限于特定的方案：
> > https://wubi98.gitee.io/2023/02/21/2023-02-21-01.rimetool/
> > WIKI：https://gitee.com/wubi98/shell/wikis/RIME%20%E7%9B%AE%E5%BD%95%E7%9B%B4%E8%BE%BE
> > 如果考虑做基于 C 或 C++ 的离线版，我可以提供一些帮助。
> 
> 我浏览了您的 repo，您说的通用版指的是不仅「Rime」也包括其他一些输入法也能利用的「通用」吗？

 schema 的节点设计是有特征的，librime 使用的是  yaml-cpp  这个项目，解析起来可能更精准。

**Comment by pdog18** [comment]
> > > 解析 yaml 目前最优解可能就是使用 https://github.com/jbeder/yaml-cpp 这个项目，大佬可以考虑做个通用版。
> > > 目前，我们的实现是基于文本模式的读改，只能局限于特定的方案：
> > > https://wubi98.gitee.io/2023/02/21/2023-02-21-01.rimetool/
> > > WIKI：https://gitee.com/wubi98/shell/wikis/RIME%20%E7%9B%AE%E5%BD%95%E7%9B%B4%E8%BE%BE
> > > 如果考虑做基于 C 或 C++ 的离线版，我可以提供一些帮助。
> > 
> > 
> > 我浏览了您的 repo，您说的通用版指的是不仅「Rime」也包括其他一些输入法也能利用的「通用」吗？
> 
> schema 的节点设计是有特征的，librime 使用的是 yaml-cpp 这个项目，解析起来可能更精准。

感谢，您说的「节点设计」的「特征」是例如`「menu/page_size」` => `「menu:{page_size: *}」` 这种吗？我正是遇到了这类问题，只不过 yaml-cpp 似乎不适用与 TypeScript 的项目，可能使用不了...

**Comment by pdog18** [comment]
我刚刚开通了 Discussion，我将这个 Issue 转过去，方便我请教您！
